src/MiddlewareFactory.php

```diff
@@ -76,11 +76,11 @@ public function prepare($middleware) : MiddlewareInterface
             return $this->pipeline(...$middleware);
         }
 
-        if (! is_string($middleware) || ! $middleware) {
-            throw Exception\InvalidMiddlewareException::forMiddleware($middleware);
+        if (is_string($middleware)) {
```

Comment:
wrong: https://3v4l.org/DIbno

Reply:
Ooops, my mistake...

I can revert and add an exclusion in phpstan. Or simply test empty string in a way that It won't complain: 

```
// many options exists depending on where we want
// to catch the error (if empty the middleware cannot be found anyway)
if (is_string($middleware) && $middleware !== '') {        
            return $this->lazy($middleware);
}
throw Exception\InvalidMiddlewareException::forMiddleware($middleware);
```

Let me know

Reply:
tbh I don't like it, this is PHP not Java, nor any strongly typed programming language, and making changes like that are covering php flexibility. I know many people will disagree with me...

Reply:
I'm not purist neither, so I agree with you 100%. The idea behind this P/R is to propose phpstan as Q&A tool.  

So I tried to figure out where it would complain and tried to minimize ignored patterns. Not to make it Java like ;) 

The goal is more Q&A for people like me who would gladly contribute from times to times. PHPstan can be a good safety belt. 

I'll check tomorrow to see what's best to do and probably revert this one.

Reply:
> The condition is replaced wrong, so not sure if the tool works correctly?

without the change phpstan 0.9 will complain about

```
------ --------------------------------------------------------------- 
  Line   src/MiddlewareFactory.php                                      
 ------ --------------------------------------------------------------- 
  79     Only booleans are allowed in a negated boolean, string given.  
 ------ --------------------------------------------------------------- 
```

`array|callable|MiddlewareInterface|RequestHandlerInterface` have been well discarded by previous tests (is_array, is_callable...). The remaining possible type is `string`... In `strict_mode` it will complain. We can choose of course